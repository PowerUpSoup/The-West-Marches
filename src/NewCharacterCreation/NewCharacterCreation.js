import React, { Component } from 'react';
import apiContext from '../ApiContext';
import config from '../config';

class NewCharacterCreation extends Component {

    static contextType = apiContext;

    handleSubmitCharacterCreate = (e) => {
        e.preventDefault();
        let sessionStorageUser = JSON.parse(sessionStorage.getItem("user"))
        const character = {
            "user_id": sessionStorageUser.id,
            "name": this.refs.characterUsername.value.toLowerCase()
        }
        fetch(`${config.API_BASE_URL}/characters`, {
            method: 'post',
            headers: { 'content-Type': 'application/json' },
            body: JSON.stringify({ user_id:character.user_id, name:character.name })
        }).then(res => {
            if (!res.ok)
                return res.json().then(e => Promise.reject(e))
            return res.json()
        })
            .then((data) => {
                this.context.addCharacter(data)
                this.props.history.push('/home')
            }).catch(error => {
                console.error({ error })
            })
        }

    render() {
        return (
            <div>
                <form onSubmit={(e) => {
                    this.handleSubmitCharacterCreate(e)
                }}>
                    <label>Character Name:</label>
                    <input type="text"
                        name="create-username"
                        placeholder="Charater Name:"
                        ref="characterUsername"
                        required />
                </form>
            </div>
        )
    }
}

export default NewCharacterCreation;