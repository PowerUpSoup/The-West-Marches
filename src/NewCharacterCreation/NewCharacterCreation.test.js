  
import React from 'react';
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json'
import NewCharacterCreation from './NewCharacterCreation';

describe(`New Character Creation component`, () => {

    it('renders a NewCharacterCreation', () => {
        const wrapper = shallow(
            <BrowserRouter>
                <NewCharacterCreation />
            </BrowserRouter>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})