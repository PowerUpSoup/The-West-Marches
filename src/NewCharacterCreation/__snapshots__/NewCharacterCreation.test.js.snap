// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Character Creation component renders a NewCharacterCreation 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <NewCharacterCreation />
</Router>
`;
